nixos/modules/system/boot/stratisroot.nix

```diff
@@ -0,0 +1,76 @@
+{ config, lib, pkgs, ... }:
+let
+  types = lib.types;
+in
+{
+  options.boot.stratis = {
+    rootPoolUuid = lib.mkOption {
+      type = types.uniq types.str;
+      description = lib.mdoc ''
+        UUID of the stratis pool that the root fs is located in
+      '';
+      example = "04c68063-90a5-4235-b9dd-6180098a20d9";
+    };
+  };
+  config = {
+    assertions = [
+      {
+        assertion = config.boot.initrd.systemd.enable;
+        message = "stratis root fs requires systemd initrd";
+      }
+      {
+        assertion = lib.strings.hasPrefix "/dev/stratis/" config.fileSystems."/".device;
+        message = "root file system must live in /dev/stratis";
+      }
+      {
+        assertion = config.boot.plymouth.enable;
+        message = "plymouth must be enabled for stratis pool password prompt";
+      }
+    ];
+    boot.initrd = {
+      systemd = {
+        storePaths = [
+          "${pkgs.stratisd}/lib/udev/stratis-base32-decode"
+          "${pkgs.stratisd}/lib/udev/stratis-str-cmp"
+          "${pkgs.lvm2.bin}/bin/dmsetup"
+          "${pkgs.stratisd}/libexec/stratisd-min"
+          "${pkgs.stratisd.initrd}/bin/stratis-rootfs-setup"
+        ];
+        packages = [pkgs.stratisd.initrd];
+        extraBin = {
+          thin_check = "${pkgs."thin-provisioning-tools"}/bin/thin_check";
+          thin_repair = "${pkgs."thin-provisioning-tools"}/bin/thin_repair";
+          thin_metadata_size = "${pkgs."thin-provisioning-tools"}/bin/thin_metadata_size";
+          stratis-min = "${pkgs.stratisd}/bin/stratis-min";
+        };
+        services = {
+          stratis-setup = {
+            description = "setup for Stratis root filesystem";
+            unitConfig.DefaultDependencies = "no";
+            conflicts = [ "shutdown.target" ];
+            onFailure = [ "emergency.target" ];
+            unitConfig.OnFailureJobMode = "isolate";
+            wants = [ "stratisd-min.service" "plymouth-start.service" "stratis-clevis-setup.service" ];
```

Comment:
Shouldn't have `plymouth-start.service` in here if plymouth isn't actually required.

What is `stratis-clevis-setup.service`?

Reply:
It is a generated unit, by `stratis-clevis-setup-generator`. mostly the same as `stratis-clevis-setup.service`, but for clevis encrypted volumes. We can safely omit it if not supporting clevis for now.

Reply:
For now, Plymouth is required because the stratis-rootfs-setup calls it directly.

I've removed the stratis-clevis-setup.service file